sdk/src/embedded_xmp.rs

```diff
@@ -22,20 +24,28 @@ use xmp_toolkit::{OpenFileOptions, XmpFile, XmpFileError, XmpMeta};
 ///
 /// This does not check the claim at all; it is presumed
 /// that the string that is passed is a valid signed claim.
-pub(crate) fn add_manifest_uri_to_file<P: AsRef<Path>>(
-    path: P,
-    manifest_uri: &str,
-) -> Result<(), XmpFileError> {
-    XmpMeta::register_namespace("http://purl.org/dc/terms/", "dcterms");
+pub(crate) fn add_manifest_uri_to_file<P: AsRef<Path>>(path: P, manifest_uri: &str) -> Result<()> {
+    XmpMeta::register_namespace("http://purl.org/dc/terms/", "dcterms").map_err(xmp_write_err)?;
+
+    let mut f = XmpFile::new().map_err(xmp_write_err)?;
+
+    f.open_file(path, OpenFileOptions::default().for_update())
+        .map_err(xmp_write_err)?;
 
-    let mut f = XmpFile::new();
+    let mut m = match f.xmp() {
```

Comment:
Does this work?
let mut m = f.xmp().unwrap_or_else(XmpMeta::new).map_err(xmp_write_err)?;

Reply:
Tried that. It would have worked if both `f.xmp()` and `XmpMeta::new` returned `Result`, but they do not.

`f.xmp()` is an `Option` because a file not having XMP is a legitimate, non-error condition. And I really don't like the ergonomics of `Result<Option<xxx>>`, so `f.xmp()` folds error conditions into a `None` response.

Reply:
ok, just checking

Reply:
How do you create an empty XmpFile or is that not possible?